vizro-core/src/vizro/models/_controls/filter.py

```diff
@@ -36,11 +36,15 @@
 }
 
 
-def _filter_between(series: pd.Series, value: List[float]) -> pd.Series:
+def _filter_between(series: pd.Series, value: Union[List[float], List[str]]) -> pd.Series:
+    if is_datetime64_any_dtype(series):
+        value = pd.to_datetime(value)
     return series.between(value[0], value[1], inclusive="both")
 
 
 def _filter_isin(series: pd.Series, value: MultiValueType) -> pd.Series:
+    if is_datetime64_any_dtype(series):
+        value = pd.to_datetime(value)
```

Comment:
I think we need the same conversion of `series` here too to ensure that all times in the given day are included.